### ISSUE TITLE
False Positive for `Rails/Delegate` Cop

### ISSUE THREAD

**Issue by pioz** [rubocop/rubocop#13877]
The `Rails/Delegate` cop currently triggers a warning "Use `delegate` to define delegations" for the operator `[]` method defined in a class. However, this warning is not applicable here since the `[]` method is used to access elements of an internal array rather than delegating behavior.

--------

## Steps to reproduce the problem

Create a class with the following code:
   ```ruby
  class Foo
     def initialize(array)
       @array = array
     end

     def [](index) # <- warning  rubocop  Use `delegate` to define delegations.
       @array[index]
     end

     def pop # <- warning  rubocop  Use `delegate` to define delegations.
       @array.pop
     end
 end
```

## RuboCop version

```
1.72.2 (using Parser 3.3.7.1, rubocop-ast 1.38.0, analyzing as Ruby 3.4, running on ruby 3.4.2) [arm64-darwin24]
  - rubocop-minitest 0.37.1
  - rubocop-performance 1.24.0
  - rubocop-rails 2.30.1
  - rubocop-capybara 2.21.0
```
